### PR TITLE
NAS-133109 / 25.04 / Refactor PAM config generation

### PR DESCRIPTION
**Pull request by anodos325**
This commit refactors how we generate custom PAM configuration lines so as to force them to run though a parser, and adds tests for said parser. pam_tty_audit is also enabled for interactive sessions for root account if for some reason a user gets interactive root session when GPOS STIG is enabled (this should never happen).

### CODE REVIEW

File: src/middlewared/middlewared/etc_files/pam.d/common-session.mako [hunk 1]
```diff
@@ -6,14 +6,36 @@
...
+            conf = AD_SESSION
+        case DSType.LDAP.value | DSType.IPA.value:
+            conf = SSS_SESSION
+        case _:
+            raise TypeError(f'{dstype}: unknown DSType')
+
+    if render_ctx['system.security.config']['enable_gpos_stig']:
+        tty_audit_line = TTY_AUDIT_LINE
+%>\
+% if conf.tty_audit_line:
```
Comment (bmeagherix) [inline src/middlewared/middlewared/etc_files/pam.d/common-session.mako:29]: Don't think this should have a leading `conf.`

**Review by bmeagherix — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/etc_files/pam.d/common-session.mako [hunk 2]
```diff
@@ -6,14 +6,36 @@
...
+        case None:
+            conf = STANDALONE_SESSION
+        case DSType.AD.value:
+            conf = AD_SESSION
+        case DSType.LDAP.value | DSType.IPA.value:
+            conf = SSS_SESSION
+        case _:
+            raise TypeError(f'{dstype}: unknown DSType')
+
+    if render_ctx['system.security.config']['enable_gpos_stig']:
```
Comment (bmeagherix) [inline src/middlewared/middlewared/etc_files/pam.d/common-session.mako:26]: Is `system.security.config` in the render context?

**Review by bmeagherix — APPROVED** [review]

File: src/middlewared/middlewared/utils/pam.py [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 2),
+                PAMControl(PAMResponse.DEFAULT, PAMAction.IGNORE)
+            ),
+            pam_module=PAMModule.UNIX
+        ),
+        PAMLine(
+            pam_service=PAMService.AUTH,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 2),
```
Comment (bmeagherix) [inline src/middlewared/middlewared/utils/pam.py:227]: I think this should be 
```
PAMControl(PAMResponse.SUCCESS, 1),
```

**Review by bmeagherix — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/utils/pam.py [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+    )
+)
+
+SSS_ACCOUNT = PAMConfLines(
+    service=PAMService.ACCOUNT,
+    primary=(
+        PAMLine(
+            pam_service=PAMService.ACCOUNT,
+            pam_control=(
+                PAMControl(PAMResponse.SUCCESS, 2),
```
Comment (mgrimesix) [inline src/middlewared/middlewared/utils/pam.py:155]: In current code, attached to the FreeIPA I see the following in `common-account`:
`account [success=1 new_authtok_reqd=done default=ignore]        pam_unix.so`
Maybe this line should have `1` and not `2`?

File: src/middlewared/middlewared/utils/pam.py (hunk 3 above)
Comment (mgrimesix) [reply src/middlewared/middlewared/utils/pam.py:227]: I agree.  `1` is what's used in current common-auth for a server attached to FreeIPA.

**Review by mgrimesix — CHANGES_REQUESTED** [review]

**Review by bmeagherix — APPROVED** [review]

**Pull request merged**